registry_module.go

```diff
@@ -252,6 +258,17 @@ type RegistryModuleUpdateOptions struct {
 	// Optional: Flag to enable no-code provisioning for the whole module.
 	// **Note: This field is still in BETA and subject to change.**
 	NoCode *bool `jsonapi:"attr,no-code,omitempty"`
+
+	// Optional: Flag to enable tests for the module
+	// **Note: This field is still in BETA and subject to change.**
+	TestConfig *RegistryModuleTestConfigOptions `jsonapi:"attr,test-config,omitempty"`
+
+	Branch string `jsonapi:"attr,branch,omitempty"`
+	Tags   string `jsonapi:"attr,tags,omitempty"`
```

Comment:
These fields are unexpected, because there's only one Update method, which acts on both VCS-backed and non-VCS-backed modules. 

Also, I can't find any API docs for the PATCH endpoint with registry modules, so I'm not sure how this corresponds to the backing API yet. 🤔 If you've got the details, that'd be helpful to know.

What's the expected behavior when you provide one of these for a non-VCS module? What's the expected behavior when you provide _both_ of these? I see you're not doing any validity checking in the client in this PR, so presumably you get whatever the backend gives ya...

At minimum these fields need some godoc comments to clarify when and how to use em.

Reply:
I didn't add any client side validation because the update method only takes the `RegistryModuleID`, which does not feature the `PublishingMechanism`, which is what would be needed to enforce client side validation on these fields. 

I can add additional documentation on how these fields operate.